R/shinyGadget.R

```diff
@@ -1,69 +1,177 @@
 projectGadget <- function() {
+  if (!requireNamespace("shinyFiles", quietly = TRUE)) {
+    stop("The shinyFiles package is required for this gadget to work. Please install it.",
+      call. = FALSE
+    )
+  }
+
   ui <- miniUI::miniPage(
-    miniUI::gadgetTitleBar("Create pRoject"),
+    miniUI::gadgetTitleBar("Create Project"),
     miniUI::miniContentPanel(
+      shiny::tags$style("#folder {width: 435px; };",
+        type = "text/css"
+      ),
       shiny::inputPanel(
         shiny::selectInput(
           "projectType",
           label = "Project Type",
-          choices = c("Basic", "Analysis", "Package", "Training"),
+          choices = c("Basic", "Package", "Analysis", "Training"),
           selected = "Basic"
         ),
-        shiny::textInput("name", "Project Name")
+        shiny::selectInput(
+          "initial_status",
+          label = "Initial Status",
+          choices = statuses()$status,
+          selected = "wip"
+        ),
+        shiny::textInput("name", "Project Name", value = NULL),
+        shiny::textInput("title", "Project Title", value = NULL),
+        shiny::div("Project Folder", class = "control-label", style = "font-weight: bold;"),
+        shiny::br(),
+        shiny::br(),
+        shiny::br(),
+        shinyFiles::shinyDirButton("folder",
+          label = "Choose Folder",
+          title = "Please choose a project folder",
+          class = "btn-primary"
+        ),
+        shiny::br(),
+        shiny::uiOutput("folder")
       ),
       shiny::uiOutput("ui"),
       shiny::conditionalPanel(
-        condition = "input.projectType != 'Package'",
+        condition = "input.externalSetup",
```

Comment:
Would a hover explanation "scroll down after checking this box" help? I expected more options to appear and scrolled down but I wonder how intuitive this is. This is my only complaint about the interface that's very nice.

Reply:
This is a good point - didn't pick up on this after switching back to displaying the gadget in the RStudio pane (obviously this wasn't an issue in the bigger dialog/modal).

We could add a tooltip here as a hint, but I think it may be better if it autoscrolls to the bottom when this checkbox is enabled to show the new config options that become available. I've just pushed an update which does this - so please try it out and see if you approve. The one downside I think is that it might be slightly jarring for some, but I think it's more obvious than a tooltip which can be missed if one doesn't keep the cursor still and in the right position for more than a second or two.

Reply:
I had missed this comment :scream: I have now tested this, I like it! It might puzzle some users indeed but I think that's the best solution, and a very cool trick.